### PR TITLE
[or1k-cluster] Move OMPIC to Cluster AL

### PR DESCRIPTION
**Pull request by Theldus**
Description
--------------
This PR moves the OMPIC device interface to the Cluster AL, where it really belongs to.

Related Issues
--------------------
[[or1k-cluster] Move OMPIC to Cluster AL](https://github.com/nanvix/hal/issues/375)

### CODE REVIEW

File: include/arch/cluster/optimsoc-cluster/ompic.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef CLUSTER_OPTIMSOC_CLUSTER_OMPIC_H_
+#define CLUSTER_OPTIMSOC_CLUSTER_OMPIC_H_
+
```
Comment (ppenna) [inline include/arch/cluster/optimsoc-cluster/ompic.h]: we should `#include <arch/cluster/optimsoc-cluster/_optimsoc-cluster.h>`

File: include/arch/cluster/or1k-cluster/ompic.h [hunk 2]
```diff
@@ -22,8 +22,8 @@
  * SOFTWARE.
  */
 
-#ifndef ARCH_CORE_OR1K_OMPIC_H_
-#define ARCH_CORE_OR1K_OMPIC_H_
+#ifndef CLUSTER_OR1K_CLUSTER_OMPIC_H_
+#define CLUSTER_OR1K_CLUSTER_OMPIC_H_
 
 	#include <nanvix/const.h>
 	#include <stdint.h>
```
Comment (ppenna) [inline include/arch/cluster/or1k-cluster/ompic.h]: we should `#include <arch/cluster/or1k-cluster/_or1k-cluster.h>`

File: include/arch/cluster/optimsoc-cluster/ompic.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	#include <nanvix/const.h>
+	#include <stdint.h>
+
+#ifndef _ASM_FILE_
+
+	/* External functions. */
+	EXTERN void or1k_ompic_init(void);
```
Comment (ppenna) [inline include/arch/cluster/optimsoc-cluster/ompic.h]: instead `#include <arch/cluster/or1k-cluster/ompic.h>` and create aliases with `optimsoc_cluster` prefix.

File: include/arch/cluster/or1k-cluster/ompic.h [hunk 4]
```diff
@@ -22,8 +22,8 @@
  * SOFTWARE.
  */
 
-#ifndef ARCH_CORE_OR1K_OMPIC_H_
-#define ARCH_CORE_OR1K_OMPIC_H_
+#ifndef CLUSTER_OR1K_CLUSTER_OMPIC_H_
+#define CLUSTER_OR1K_CLUSTER_OMPIC_H_
```
Comment (ppenna) [inline include/arch/cluster/or1k-cluster/ompic.h]: we should rename functions from `or1k_*` to `or1k_cluster_*`

**Comment by Theldus** [comment]
@ppenna Could you re-check again and see if everything is as expected?
I will deal with the cluster aliases and separation in another PR, ;-).

**Comment by ppenna** [comment]
@Theldus good work =)

**Pull request merged**